### PR TITLE
refactor: 💡 Convert SQFormCheckboxGroup to typescript Ref #244

### PR DESCRIPTION
**Pull request by caleboleary**
✅ Closes: #315

Loom of it functioning in storybook: https://www.loom.com/share/6bcbc4de7364403eb49ec08f6d3d084a

### CODE REVIEW

File: src/components/SQForm/SQFormCheckboxGroup.tsx [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
 import FormHelperText from '@material-ui/core/FormHelperText';
 import FormGroup from '@material-ui/core/FormGroup';
+import {CheckboxProps} from '@material-ui/core';
 import SQFormCheckboxGroupItem from './SQFormCheckboxGroupItem';
 import SQFormCheckbox from './SQFormCheckbox';
 import {useForm} from './useForm';
 
+interface CheckboxOption {
+  label: string;
+  value: unknown;
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormCheckboxGroup.tsx:15]: I'm pretty sure, like our dropdown options, we only allow number, string, or boolean for value. maybe not though 🤔

File: src/components/SQForm/SQFormCheckboxGroup.tsx [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+  isDisabled?: boolean;
+  inputProps?: CheckboxProps;
+}
+
+interface SQFormCheckboxGroupProps {
+  /** Name of the checkbox group */
+  name: string;
+  /** Label to display above the group */
+  groupLabel: string;
+  /** Whether this a selection in this group is required */
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormCheckboxGroup.tsx:25]: nit: typo in this description

File: src/components/SQForm/SQFormCheckboxGroup.tsx [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
+  isRequired?: boolean;
+  /** Function to call on value change */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Whether to display the group in a row */
+  shouldDisplayInRow?: boolean;
+  /** Whether to display the select all checkbox */
+  shouldUseSelectAll?: boolean;
+  /** Size of the input given full-width is 12. */
+  size?: 'auto' | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormCheckboxGroup.tsx:34]: MUI provides a GridSize type. It's used in the `BaseFieldProps`

File: src/components/SQForm/SQFormCheckboxGroup.tsx (hunk 1 above)
Comment (caleboleary) [reply src/components/SQForm/SQFormCheckboxGroup.tsx:15]: Ok, I'll try to narrow it and see if it breaks anything

**Review by 20BBrown14 — APPROVED** [review]

**Review by codyarose — APPROVED** [review]

**Pull request merged**